extensions/parquet/base/src/main/java/io/deephaven/parquet/base/RleIntChunkedWriter.java

```diff
@@ -95,22 +97,32 @@ public String memUsageString(String prefix) {
     }
 
     @Override
-    public void writeBulk(@NotNull IntBuffer bulkValues, int rowCount) {
-
+    public void writeBulk(@NotNull IntBuffer bulkValues,
+                          final int rowCount,
+                          @Nullable final Statistics<?> statistics) {
+        // Track statistics while we write the values.
         for (int i = 0; i < rowCount; i++) {
-            writeInteger(bulkValues.get());
+            final int v = bulkValues.get();
+            writeInteger(v);
+            statistics.updateStats(v);
```

Comment:
From what I understood, if we use dictionary encoding, the statistics will be collected while building the dictionary in ParquetTableWriter. If we use plain encoding, they will be populated here. 

Is there any reason why we don't want to keep statistics logic just in the ChunkedWriters? Is it because the data that reaches RleIntChunkedWriter is actually indexes into the dictionary and not actual values? If that is the reason, do we need statistics counting logic in RleIntChunkedWriter?

Reply:
I may misunderstand, but RleIntChunkedWriter is currently used by DHC only for dictionary values, but it can also be used for any integer column as well.

Reply:
Yea makes sense. Thanks for explaining.